src/net.rs

```diff
@@ -359,7 +364,16 @@ impl NetworkStream for HttpStream {
 
 /// A connector that will produce HttpStreams.
 #[derive(Debug, Clone, Default)]
-pub struct HttpConnector;
+pub struct HttpConnector {
+    pub proxy: Option<Proxy>
```

Comment:
what if a `ProxyConnector` existed in `src/proxy.rs`, that could wrap any `C: NetworkConnector`? This would allow all custom connectors to be wrapped, as well as let the logic stay in 1 module. It would also remove the `if proxy.is_some()` branch from code that never used a proxy.